packages/react-instantsearch/src/connectors/connectGeoSearch.js

```diff
@@ -0,0 +1,228 @@
+
+const currentRefinementToString = currentRefinement =>
+  [
+    currentRefinement.northEast.lat,
+    currentRefinement.northEast.lng,
+    currentRefinement.southWest.lat,
+    currentRefinement.southWest.lng,
+  ].join();
+
+const stringToCurrentRefinement = value => {
+  const values = value.split(',');
+
+  return {
+    northEast: {
+      lat: parseFloat(values[0]),
+      lng: parseFloat(values[1]),
+    },
+    southWest: {
+      lat: parseFloat(values[2]),
+      lng: parseFloat(values[3]),
+    },
+  };
+};
+
+const latLngRegExp = /^(-?\d+(?:\.\d+)?),\s*(-?\d+(?:\.\d+)?)$/;
+const stringToPosition = value => {
+  const pattern = value.match(latLngRegExp);
+
+  return {
+    lat: parseFloat(pattern[1]),
+    lng: parseFloat(pattern[2]),
+  };
+};
+
+const getCurrentRefinement = (props, searchState, context) => {
+  const refinement = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    getBoundingBoxId()
+  );
+
+  if (!refinement) {
+    return refinement;
+  }
+
+  return {
+    northEast: {
+      lat: parseFloat(refinement.northEast.lat),
+      lng: parseFloat(refinement.northEast.lng),
+    },
+    southWest: {
+      lat: parseFloat(refinement.southWest.lat),
+      lng: parseFloat(refinement.southWest.lng),
+    },
+  };
+};
+
+const getCurrentPosition = (props, searchState, context) => {
+  const aroundLatLng = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    getAroundLatLngId()
+  );
+
+  if (!aroundLatLng) {
+    // Fallback on `configure.aroundLatLng`
+    const configureAroundLatLng = getCurrentRefinementValue(
+      props,
+      searchState,
+      context,
+      getConfigureAroundLatLngId()
+    );
+
+    return configureAroundLatLng && stringToPosition(configureAroundLatLng);
+  }
+
+  return aroundLatLng;
+};
+
+const refine = (searchState, nextValue, context) => {
+  const resetPage = true;
+  const nextRefinement = {
+    [getBoundingBoxId()]: nextValue,
+  };
+
+  return refineValue(searchState, nextRefinement, context, resetPage);
+};
+
+export default createConnector({
+  displayName: 'AlgoliaGeoSearch',
+
+  getProvidedProps(props, searchState, searchResults) {
+    const results = getResults(searchResults, this.context);
+
+    // We read it from both becuase the SearchParameters & the searchState are not always
+    // in sync. When we set the refinement the searchState is used when we clear the refinement
+    // the SearchParameters is used. In the first case when we render the results are not there
+    // so we can't find the value from the results. The most up to date value is the searchState.
+    // But whren we clear the refinement the searchState is immediatly clear even when the items
+    // retrieve are still the one from the previous query with the bounding box. It leads to some
+    // issue with the position of the map. We should rely on 1 source of truth or at least always
+    // be sync.
```

Comment:
Yes, in all other connectors we never rely on the helper state only on the `searchState`. In that case we can't because both states are not always sync. Most of the time it's not an issue but in case of the geo search it leads to inconsistencies for the view.